### PR TITLE
Revert validation of operation signature versioning

### LINKED ISSUE #2688: "incompatible versioned reference" in function signature
Introduced by #2684.

```
run command:npx --no tsp compile client.tsp --no-emit --warn-as-error
TypeSpec compiler v0.51.0-dev.7

Diagnostics were reported during compilation:

/mnt/vss/_work/1/s/specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp:87:3
- error @typespec/versioning/incompatible-versioned-reference: 'Azure.OpenAI.getChatCompletionsWithAzureExtensions'
was added in version 'v2023_08_01_Preview' but contains type 'Azure.OpenAI.(anonymous model).functions' added in
version 'v2023_07_01_Preview'.
> 87 |   functions?: FunctionDefinition[];
     |   ^^^^^^^^^
```
```
run command:npx --no tsp compile client.tsp --no-emit --warn-as-error
TypeSpec compiler v0.51.0-dev.7

Diagnostics were reported during compilation:

/mnt/vss/_work/1/s/specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:315:6
- error @typespec/versioning/incompatible-versioned-reference: 'Microsoft.EventGrid.releaseCloudEvents' is referencing
versioned type 'Microsoft.EventGrid.(anonymous model).releaseDelayInSeconds' but is not versioned itself.
> 315 |   op releaseCloudEvents is StandardOperations.ResourceAction<
      |      ^^^^^^^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by tjprescott**
Fix #2688.

Azure Core PR: https://github.com/Azure/typespec-azure/pull/3953

### CODE REVIEW

File: packages/versioning/test/incompatible-versioning.test.ts [hunk 1]
```diff
@@ -103,51 +104,43 @@ describe("versioning: validate incompatible references", () => {
   });
 
   describe("operation", () => {
-    it("emit diagnostic when unversioned op has a versioned model as a parameter", async () => {
-      const diagnostics = await runner.diagnose(`
+    it("allow unversioned op to have versioned model as a parameter", async () => {
```
Comment (timotheeguerin) [inline packages/versioning/test/incompatible-versioning.test.ts:107]: I think the test before was correct, if model `Foo` was added in v2 there shoulnd't be a way to use in in v1

File: packages/versioning/test/incompatible-versioning.test.ts [hunk 2]
```diff
@@ -103,51 +104,43 @@ describe("versioning: validate incompatible references", () => {
...
     });
 
-    it("emit diagnostic when unversioned op has a versioned parameter", async () => {
-      const diagnostics = await runner.diagnose(`
+    it("allow unversioned op to have a versioned parameter", async () => {
+      ok(
+        await runner.compile(`
         model Foo {}
 
         op test(param: string, @added(Versions.v2) newParam: Foo): void;
```
Comment (timotheeguerin) [inline packages/versioning/test/incompatible-versioning.test.ts:123]: This one I think the change is correct

File: packages/versioning/test/incompatible-versioning.test.ts [hunk 3]
```diff
@@ -103,51 +104,43 @@ describe("versioning: validate incompatible references", () => {
...
-        message:
-          "'TestService.test' is referencing versioned type 'TestService.(anonymous model).newParam' but is not versioned itself.",
-      });
+      `)
+      );
     });
 
-    it("emit diagnostic when unversioned op based on a template has a versioned model as a parameter", async () => {
-      const diagnostics = await runner.diagnose(`
+    it("allow unversioned op based on a template to have a versioned model as a parameter", async () => {
```
Comment (timotheeguerin) [inline packages/versioning/test/incompatible-versioning.test.ts:128]: This one I think was correct before, it should emit an error

**Comment by tjprescott** [comment]
@timotheeguerin I added `skip` to the tests as you described and linked them to the implementation issue for operation signature validation.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**